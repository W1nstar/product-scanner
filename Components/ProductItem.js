import React, { Component } from 'react'
import { StyleSheet, View, Text, Image } from 'react-native'
import PropTypes from "prop-types";

class ProductItem extends Component {
    render() {
        const product = this.props.product;
        const cartCounter = this.props.cartCounter;
        return (
            <View style={styles.mainContainer}>
                <Image
                    style={styles.image}
                    source={{uri: product.imageUrl}}
                />
                <View style={styles.contentContainer}>
                    <View style={styles.headerContainer}>
                        <Text style={styles.titleText}>{product.name}</Text>
                        <Text style={styles.voteText}>{cartCounter}</Text>
                    </View>
                    <View style={styles.descriptionContainer}>
                        <Text style={styles.descriptionText}>
                            {product.categories !== undefined && product.categories.length > 0 
                                ? product.categories[product.categories.length - 1].trim() 
                                : 'Pas de catégorie renseignée'}
                        </Text>
                        {
                            cartCounter === undefined  // history view
                                ? <Text style={styles.descriptionText}>Scanné : {product.nbScans} fois</Text>
                                : null
                        }
                    </View>
                </View>
            </View>
        )
    }
}

ProductItem.propTypes = {
    product: PropTypes.shape({
        imageUrl: PropTypes.string,
        name: PropTypes.string.isRequired,
        barCode: PropTypes.string.isRequired,
        categories: PropTypes.arrayOf(PropTypes.string),
        nbScans: PropTypes.number.isRequired,
    }).isRequired,
    cartCounter: PropTypes.number,
};

export default ProductItem;

const styles = StyleSheet.create({
    mainContainer: {
        height: 190,
        flexDirection: 'row'
    },
    image: {
        width: 120,
        height: 180,
        margin: 5
    },
    contentContainer: {
        flex: 1,
        margin: 5
    },
    headerContainer: {
        flex: 3,
        flexDirection: 'row'
    },
    titleText: {
        fontWeight: 'bold',
        fontSize: 20,
        flex: 1,
        flexWrap: 'wrap',
        paddingRight: 5
    },
    voteText: {
        fontWeight: 'bold',
        fontSize: 20,
        color: '#666666'
    },
    descriptionContainer: {
        flex: 7
    },
    descriptionText: {
        fontStyle: 'italic',
        color: '#666666'
    },
    dateContainer: {
        flex: 1
    },
    dateText: {
        textAlign: 'right',
        fontSize: 14
    }
});