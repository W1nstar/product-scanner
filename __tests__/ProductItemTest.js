import React from 'react';
import ProductItem from '../Components/ProductItem';

import renderer from 'react-test-renderer';
import product from '../Helper/testData';

test('ProductItem renders correctly', () => {
    const tree = renderer.create(<ProductItem product={product}/>).toJSON();
    expect(tree).toMatchSnapshot();
});