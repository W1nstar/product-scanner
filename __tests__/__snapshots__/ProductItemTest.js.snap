// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 190,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://static.openfoodfacts.org/images/products/325/798/025/2050/front_fr.7.400.jpg",
      }
    }
    style={
      Object {
        "height": 180,
        "margin": 5,
        "width": 120,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
            "fontSize": 20,
            "fontWeight": "bold",
            "paddingRight": 5,
          }
        }
      >
        Oeufs bio nature x6
      </Text>
      <Text
        style={
          Object {
            "color": "#666666",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 7,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#666666",
            "fontStyle": "italic",
          }
        }
      >
        Œufs de poules élevées en plein air
      </Text>
      <Text
        style={
          Object {
            "color": "#666666",
            "fontStyle": "italic",
          }
        }
      >
        Scanné : 
        1
         fois
      </Text>
    </View>
  </View>
</View>
`;
